### PR TITLE
Tailwind CSS Migration: PhoButton component

### LINKED ISSUE #102: Tailwind CSS Migration: PhoButton component
Style `src/vue/components/ui/forms/PhoButton.vue` to look tidy.

### PR DESCRIPTION
**Pull request by yurianxdev**
Basic styling on button component `PhoButton`, Fixes #102

### CODE REVIEW

File: src/vue/components/ui/PhoButton.vue [hunk 1]
```diff
@@ -1,12 +1,25 @@
 <template>
-  <button type="button" @click="onClick"><slot /></button>
+  <button
+    :class="{'bg-gray-300 text-gray-900': disabled,
```
Comment (dipasqualew) [inline src/vue/components/ui/PhoButton.vue:3]: I think we are likely to use several variances of the `PhoButton`, like `primary`, `info`, `error`. It would be easier to move the classes into the `setup` function and bind an object instead? What do you think?

File: src/vue/components/pages/services/ServiceForm.vue [hunk 2]
```diff
@@ -6,19 +6,19 @@
     </div>
     <div class="grid grid-cols-1 sm:grid-cols-2 gap-4">
       <Integration :service="service" selected />
-      <div>
-        <div>
-          <PhoInputText label="Photion Username" v-model="user.name" cy="field:user.name" />
-          <PhoInputText label="Encryption Password" v-model="user.password" type="password" cy="field:user.password" />
+      <div class="space-y-4">
+        <div class="space-y-4">
```
Comment (dipasqualew) [inline src/vue/components/pages/services/ServiceForm.vue:10]: Will we need to add these divs to each button?

**Review by dipasqualew — COMMENTED** [review]
The button looks very nice! The font reminds me of Comic Sans but that can be changed easily.

I left a couple of questions, but overall well done. Thanks!

File: src/vue/components/pages/services/ServiceForm.vue (hunk 2 above)
Comment (yurianxdev) [reply src/vue/components/pages/services/ServiceForm.vue:10]: Not really, just added those to better looking on that form

File: src/vue/components/ui/PhoButton.vue (hunk 1 above)
Comment (yurianxdev) [reply src/vue/components/ui/PhoButton.vue:3]: How do you think this could be achieved? With a prop of type string? Parsing it inside setup and assigning the corresponding class?

Comment (dipasqualew) [reply src/vue/components/ui/PhoButton.vue:3]: I think you can have an object with this shape:

const css = computed(() => {
  myClass1: boolCondition1,
  myClass2: boolCondition2,
  myClass3: boolCondition3,
});

return {
  css,
}
```
Then you can reference it as
```
<button :class="css" />
```

At least it used to work on Vue 2 - I read nothing against this practice.

**Comment by yurianxdev** [comment]
@dipasqualew take a look at how this merge looks like lol

**Comment by dipasqualew** [comment]
It's only the last commit right? Then the easiest thing to do is to cherry pick the commit in a rebased branch

**Comment by yurianxdev** [comment]
Ready for review @dipasqualew  :)

File: src/vue/components/ui/PhoButton.vue [hunk 3]
```diff
@@ -1,19 +1,43 @@
...
+        return ['primary', 'secondary', 'warning', 'danger', 'disabled'].indexOf(color) !== -1;
+      },
+    },
+  },
+  setup(props, context) {
     const onClick = ($event: MouseEvent) => {
       return context.emit('click', $event);
     };
 
+    const cssButton = computed(() => ({
```
Comment (dipasqualew) [inline src/vue/components/ui/PhoButton.vue:30]: I think this is a bit hard to parse and possibly hard to extend...

What are your thoughts about:

```
const getColorClasses = (color: string) => {
  switch (color) {
  case 'primary':
    return 'bg-gray-800 text-white hover:shadow-lg hover:bg-gray-700'
  case 'secondary'
    ...
  }
}

const cssButton = computed(() => {
  const colorClasses = getColorClasses(props.color);

  return {
    [colorClasses]: true
  }
}
```

**Review by dipasqualew — APPROVED** [review]
Well done! Also solving that typing issue 🙂 

I left a suggestion in case you like it, but I am approving the PR in either case. Just pick the solution you prefer!

File: src/vue/components/ui/PhoButton.vue (hunk 3 above)
Comment (yurianxdev) [reply src/vue/components/ui/PhoButton.vue:30]: I think would be almost the same, no? That function only will take the complexity in another place, not reducing it.

Comment (dipasqualew) [reply src/vue/components/ui/PhoButton.vue:30]: The performance gain is quite small (we evaluate only 1 condition, but then we call a function...) however I think it's easier to read. If you feel that readability is about the same, it's probably just me, so go ahead with the current

Comment (yurianxdev) [reply src/vue/components/ui/PhoButton.vue:30]: Maybe at some point, we will need a button with more complex functionality, so I propose that when that time comes, we could refactor the button based on that. I think is not necessary to add complexity so far :)
